tests/test_encoding.py

```diff
@@ -410,6 +414,74 @@ def test_encode_nrc_const(self) -> None:
             # NRC-CONST parameters cannot be directly specified
             resp.encode(param2=0xEF, param3=0xAB)
 
+    def test_encode_system_parameter(self) -> None:
+        diag_coded_type = StandardLengthType(
+            base_data_type=DataType.A_UINT32,
+            base_type_encoding=None,
+            bit_length=16,
+            bit_mask=None,
+            is_highlow_byte_order_raw=None,
+            is_condensed_raw=None,
+        )
+        dop = DataObjectProperty(
+            odx_id=OdxLinkId("dop.year", doc_frags),
+            oid=None,
+            short_name="dop_year_sn",
+            long_name=None,
+            description=None,
+            admin_data=None,
+            diag_coded_type=diag_coded_type,
+            physical_type=PhysicalType(DataType.A_UINT32, display_radix=None, precision=None),
+            compu_method=IdenticalCompuMethod(
+                category=CompuCategory.IDENTICAL,
+                compu_internal_to_phys=None,
+                compu_phys_to_internal=None,
+                internal_type=DataType.A_UINT32,
+                physical_type=DataType.A_UINT32),
+            unit_ref=None,
+            sdgs=[],
+            internal_constr=None,
+            physical_constr=None,
+        )
+        param1 = SystemParameter(
+            oid=None,
+            short_name="year_param",
+            long_name=None,
+            description=None,
+            semantic=None,
+            dop_ref=OdxLinkRef.from_id(dop.odx_id),
+            dop_snref=None,
+            sysparam="YEAR",
```

Comment:
sysparam or sys_param ?

Reply:
the ODX spec calls the attribute for the `SYSTEM` tag `sysparam`...